### ISSUE TITLE
About reproducing

### ISSUE THREAD

**Issue by qiweili00** [tdemin16/multi-lane#2]
Hi, thanks for your nice work!
I tried to reproduce the results, but the reproduced results are lower than the reported results :
The reproduced results on VOCB0-C4
[Average performances till task 1]	mAP: 19.9472	amAP: 19.9472	oF1: 97.8003	cF1: 97.7120	Loss: 0.5626
[Average performances till task 2]	mAP: 37.9262	amAP: 28.9367	oF1: 75.4789	cF1: 82.5078	Loss: 0.5830
[Average performances till task 3]	mAP: 52.4028	amAP: 36.7587	oF1: 65.5625	cF1: 72.1207	Loss: 0.5473
[Average performances till task 4]	mAP: 67.1338	amAP: 44.3525	oF1: 62.5262	cF1: 68.6237	Loss: 0.5449
[Average performances till task 5]	mAP: 80.6665	amAP: 51.6153	oF1: 56.1719	cF1: 60.8502	Loss: 0.5637
The reproduced results on VOCB10-C2
[Average performances till task 1]	mAP: 48.5870	amAP: 48.5870	oF1: 92.0200	cF1: 92.4607	Loss: 0.3687
[Average performances till task 2]	mAP: 56.2140	amAP: 52.4005	oF1: 70.0420	cF1: 82.6810	Loss: 0.4759
[Average performances till task 3]	mAP: 60.7193	amAP: 55.1734	oF1: 54.5455	cF1: 73.1694	Loss: 0.8835
[Average performances till task 4]	mAP: 66.2093	amAP: 57.9324	oF1: 54.9384	cF1: 68.6736	Loss: 0.9063
[Average performances till task 5]	mAP: 66.9904	amAP: 59.7440	oF1: 45.6861	cF1: 60.9281	Loss: 1.7087
[Average performances till task 6]	mAP: 69.2228	amAP: 61.3238	oF1: 38.5900	cF1: 55.6505	Loss: 2.5716
Could you please help me to resolve the issue? 
Thank you!
My python and pytorch version is python=3.9.0 and pytorch=1.11.0+cu113

**Comment by tdemin16** [comment]
Hi, thanks for taking an interest in our work!
Sorry for having taken so long to answer you 😢 
I freshly cloned the repository, created a new conda environment with Python 3.11.17, and installed the provided requirements.
I noticed the default value for the normalization was wrongly set to "True" instead of "pre-head", so that was an error and I fixed it.
I also noticed that the numpy version automatically downloaded changed from the last time I checked and it raises an error with timm, so I specified a numpy version in the requirements to ensure it works immediately.
Now everything should be working properly.
Could you pull the last commit and retry?

Thomas

**Comment by qiweili00** [comment]
Thanks for your detailed reply. 
I can reproduce the results with the following two modifications:
1. Change normalization to "pre-head"
2. I find that the mean_average_precision function seems to calculate the mAP of seen classes and unseen classes. I replace it with 
`mask = [c for t in list(range(task_id+1)) for c in class_mask[t]] `
`mAP = utils.mean_average_precision(predictions, targets,average='none')[mask]`
    `mAP = mAP.mean()`
to only calculate the seen classes.

After these modifications, I can reproduce the results!

**Comment by tdemin16** [comment]
Perfect!
However, it seems strange to me that you had to mask the output by yourself.
Logits and targets are already masked in `evaluate_till_now_multi` (lines 244-246).

Can you check what happens to logits and targets after those two lines?

**Comment by qiweili00** [comment]
I retry the code with the package version provided in requirement.txt and reproduce the results without the second modification.
It seems that the torchmetrics version has an impact
(my previous torchmetrics version is 1.4.1)

**Comment by tdemin16** [comment]
Thanks!
Hope you managed to solve all your issues! If you have any other problems, let me know!
